packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts

```diff
@@ -0,0 +1,283 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable valid-jsdoc */
+import {getOSNameInternal} from '@webex/internal-plugin-metrics';
+import {BrowserDetection} from '@webex/common';
+import uuid from 'uuid';
+import {merge} from 'lodash';
+import {
+  anonymizeIPAddress,
+  clearEmptyKeysRecursively,
+  userAgentToString,
+} from './call-diagnostic-metrics.util';
+import {CLIENT_NAME} from '../config';
+import {ClientEvent} from './generated-types-temp/ClientEvent';
+import {RecursivePartial, Event, ClientType, SubClientType, NetworkType} from '../metrics.types';
+import CallDiagnosticEventsBatcher from './call-diagnostic-metrics-batcher';
+
+const {getOSVersion, getBrowserName, getBrowserVersion} = BrowserDetection();
+
+type GetOriginOptions = {
+  clientType: ClientType;
+  subClientType: SubClientType;
+  networkType?: NetworkType;
+};
+
+type GetIdentifiersOptions = {
+  meeting?: any;
+  mediaConnections?: any[];
+};
+
+export type SubmitClientEventOptions = {
+  meetingId?: string;
+  mediaConnections?: any[];
+  error?: any;
+  showToUser?: boolean;
+};
+
+/**
+ * @description Util class to handle Call Analyzer Metrics
+ * @export
+ * @class CallDiagnosticMetrics
+ */
+export default class CallDiagnosticMetrics {
+  meetingCollection: any;
+  webex: any;
+  // @ts-ignore
+  private callDiagnosticEventsBatcher: CallDiagnosticEventsBatcher;
+
+  /**
+   * Constructor
+   * @constructor
+   * @public
+   */
+  constructor() {
+    this.meetingCollection = null;
+  }
+
+  /**
+   * Initializes the CallDiagnosticMetrics singleton with a meeting Collection.
```

Comment:
this comment says "singleton" but I don't see anything in CallDiagnosticMetrics class implementation that makes it a singleton. It's only the parent Metrics class that's done as a singleton, so I think here we should just say "the CallDiagnosticMetrics class"

Reply:
Yeah, this is an old comment, I've fixed it. The CallDiagnosticMetrics is not a singleton anymore.